frontend/src/components/comp-top-table-bar/comp-top-table-bar.tsx

```diff
@@ -17,13 +17,15 @@ export const CompTopTableBar = (): React.ReactElement => {
   const { data, isLoading } = useGetComparisonCarsQuery();
   const [clearTable] = useClearComparisonMutation();
   const [isCleared, setIsCleared] = useState(false);
-  const ref = useRef<HTMLInputElement>();
 
-  const cars = data?.slice().sort((a, b) => a.position - b.position);
-  const ids = cars?.map((car) => car.id);
+  const cars = useMemo(
+    () => data?.slice().sort((a, b) => a.position - b.position),
+    [data],
+  );
+  const carsIds = cars?.map((car) => car.id);
   const initialData = {
     cars,
-    carsPositions: ids,
+    carsPositions: carsIds,
   };
```

Comment:
You should use `useMemo` for `carsIds` too. Because new array will be created on each render